File: OrchidCore/src/main/java/com/eden/orchid/impl/themes/tags/LogTag.java
package com.eden.orchid.impl.themes.tags;

import com.caseyjbrooks.clog.Clog;
import com.caseyjbrooks.clog.IClog;
import com.eden.common.util.EdenUtils;
import com.eden.orchid.api.compilers.TemplateTag;
import com.eden.orchid.api.options.annotations.Description;
import com.eden.orchid.api.options.annotations.Option;
import com.eden.orchid.api.options.annotations.StringDefault;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public final class LogTag extends TemplateTag {

    @Option @StringDefault("verbose")
    @Description("The priority with which to log the message.")
    public String level;

    @Option
    @Description("An options tag to display with the message.")
    public String tag;

    @Inject
    public LogTag() {
        super("log", true, false);
    }

    @Override
    public String[] parameters() {
        return new String[] {"level", "tag"};
    }

    @Override
    public void onPostExtraction() {
        IClog clog = (!EdenUtils.isEmpty(tag)) ? Clog.tag(tag) : Clog.getInstance();

        switch (level.toLowerCase()) {
            case "verbose": clog.v(getContent()); break;
            case "debug": clog.d(getContent()); break;
            case "info": clog.i(getContent()); break;
            case "warning": clog.w(getContent()); break;
            case "error": clog.e(getContent()); break;
            case "fatal": clog.wtf(getContent()); break;
            default: clog.v(getContent()); break;
        }
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/tags/LogTag.kt
package com.eden.orchid.impl.themes.tags

import com.caseyjbrooks.clog.Clog
import com.eden.common.util.EdenUtils
import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.compilers.TemplateTag
import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.options.annotations.Option
import com.eden.orchid.api.options.annotations.StringDefault
import com.eden.orchid.api.theme.pages.OrchidPage
import javax.inject.Inject

@Description(value = "Print the tag content to the console for debugging templates.", name = "Log")
class LogTag @Inject
constructor() : TemplateTag("log", TemplateTag.Type.Content, false) {

    @Option
    @StringDefault("verbose")
    @Description("The priority with which to log the message.")
    lateinit var level: String

    @Option
    @Description("An options tag to display with the message.")
    lateinit var tag: String

    override fun parameters(): Array<String> {
        return arrayOf("level", "tag")
    }

    override fun onRender(context: OrchidContext?, page: OrchidPage?) {
        val clog = if (!EdenUtils.isEmpty(tag)) Clog.tag(tag) else Clog.getInstance()

        when (level.toLowerCase()) {
            "verbose" -> clog.v(content)
            "debug" -> clog.d(content)
            "info" -> clog.i(content)
            "warning" -> clog.w(content)
            "error" -> clog.e(content)
            "fatal" -> clog.wtf(content)
            else -> clog.v(content)
        }
    }
}
